subprojects/platform-jvm/src/main/java/org/gradle/api/java/archives/internal/DefaultManifestMergeDetails.java

```diff
@@ -55,6 +57,18 @@ public String getMergeValue() {
 
     @Override
     public String getValue() {
+        if (value instanceof Provider) {
+            DeprecationLogger.deprecateMethod(ManifestMergeDetails.class, "getValue()")
+                .withAdvice("Please use #getRawValue() instead.")
+                .willBeRemovedInGradle8()
```

Comment:
Instead of deprecating, we might want to change this to `willbecomeAnErrorInGradle8`.

@big-guy WDYT?

Reply:
You are absolutely right Sterling. After some more investigation I realise now what confused me. 

It was the observation that when you have no merging going on (ie. no template manifest file to merge with), then what comes out of `DefaultManifest.getEffectiveManifestInternal(...)` contain `Providers` and not materialised values. And then I thought that if we force the materialisation sooner, that's going to be an early evaluation. 

But what I failed to notice was that the `Providers` coming out of `DefaultManifest.getEffectiveManifestInternal(...)` are being made to materialise their values right after it, so it doesn't matter if we do that a bit sooner in the code.

So yes, your fix is much simpler, makes more sense and solves the problem. I'll adopt it.